File: ThirtyFour.UI/ThirtyFour.UI.Tests/Tests/FormElementTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using ThirtyFour.UI.Form;

namespace ThirtyFour.UI.Tests.Tests
{
    [TestClass]
    public class FormElementTests : BaseTestSuite
    {
        public class FormElementImpl : FormElement
        {
            public FormElementImpl(IWebElement element) : base(element)
            {
            }
        }

        [TestMethod]
        public void ValuePropertyReturnsCorrectValue()
        {
            IWebElement el = driver.FindElement(By.Id("rorange"));
            FormElement radio = new FormElementImpl(el);
            Assert.AreEqual("orange", radio.Value);

            IWebElement inputTextEl = driver.FindElement(By.Name("input-text"));
            FormElement inputText = new FormElementImpl(inputTextEl);
            Assert.AreEqual("initial input-text value", inputText.Value);
            inputTextEl.Clear();
            inputTextEl.SendKeys("new input-text value");
            Assert.AreEqual("new input-text value", inputText.Value);

            IWebElement inputHiddenEl = driver.FindElement(By.Name("input-hidden"));
            FormElement inputHidden = new FormElementImpl(inputHiddenEl);
            Assert.AreEqual("initial input-hidden value", inputHidden.Value);
            
        }

        [TestMethod]
        public void LabelTextWhenWrapped()
        {
            IWebElement el = driver.FindElement(By.Id("rorange"));
            FormElement radio = new FormElementImpl(el);
            Assert.AreEqual("Orange", radio.LabelText);
        }

        [TestMethod]
        public void LabelTextWithForAttribute()
        {
            IWebElement el = driver.FindElement(By.Id("rviolet"));
            FormElement radio = new FormElementImpl(el);
            Assert.AreEqual("Violet", radio.LabelText);
        }

    }
}

File: java/src/test/java/com/github/danhumphrey/thirtyfour/ui/test/form/FormElementTests.java
package com.github.danhumphrey.thirtyfour.ui.test.form;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.github.danhumphrey.thirtyfour.ui.form.FormElement;
import com.github.danhumphrey.thirtyfour.ui.test.BaseTestSuite;

public class FormElementTests extends BaseTestSuite {

	 public class FormElementImpl extends FormElement {
         public FormElementImpl(WebElement element) {
        	 super(element);
         }
     }
	 

	@Test
    public void testGetValueReturnsCorrectValue() {
        WebDriver driver = getDriver();
        
		WebElement el = driver.findElement(By.id("rorange"));
        FormElement radio = new FormElementImpl(el);
        Assertions.assertEquals("orange", radio.getValue());

        WebElement inputTextEl = driver.findElement(By.name("input-text"));
        FormElement inputText = new FormElementImpl(inputTextEl);
        Assertions.assertEquals("initial input-text value", inputText.getValue());
        inputTextEl.clear();
        inputTextEl.sendKeys("new input-text value");
        Assertions.assertEquals("new input-text value", inputText.getValue());

        WebElement inputHiddenEl = driver.findElement(By.name("input-hidden"));
        FormElement inputHidden = new FormElementImpl(inputHiddenEl);
        Assertions.assertEquals("initial input-hidden value", inputHidden.getValue());

    }
	
	@Test
    public void testGetLabelTextWhenWrapped() {
		WebDriver driver = getDriver();
        WebElement el = driver.findElement(By.id("rorange"));
        FormElement radio = new FormElementImpl(el);
        Assertions.assertEquals("Orange", radio.getLabelText());
    }

	
    @Test
    public void testGetLabelTextWithForAttribute()
    {
    	WebDriver driver = getDriver();
        WebElement el = driver.findElement(By.id("rviolet"));
        FormElement radio = new FormElementImpl(el);
        Assertions.assertEquals("Violet", radio.getLabelText());
    }
	
}
